File: src/CourseProject.Backend/CourseProject.Domain/Entities/Question.cs
using CourseProject.Domain.Common.Entities;
using CourseProject.Domain.Enums;

namespace CourseProject.Domain.Entities;

public class Question : AuditableEntity
{
    public Guid TemplateId { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public QuestionType Type { get; set; }
    public IEnumerable<string> Options { get; set; }
    public bool DisplayInResults { get; set; }
    public Template Template { get; set; }
}

File: src/course-project-frontend/course-project-frontend/src/models/Question.js
class Question {
    constructor(id, templateId, title, description, type, options, displayInResults) {
        this.id = id;
        this.templateId = templateId;
        this.title = title;
        this.description = description;
        this.type = type;
        this.options = options;
        this.displayInResults = displayInResults;
    }
}

export default Question;
